### PR TITLE
Optimize EVM stack usage

### PR DESCRIPTION
**Pull request by jangko**
Although minuscule, but the improvement is visible. The GST now done in 51s from previous 52s.

### CODE REVIEW

File: nimbus/evm/interpreter/op_handlers/oph_arithmetic.nim [hunk 1]
```diff
@@ -39,35 +39,56 @@ func slt(x, y: UInt256): bool =
...
 proc addOp (k: var VmCtx): EvmResultVoid =
   ## 0x01, Addition
-  let (lhs, rhs) = ? k.cpt.stack.popInt(2)
-  k.cpt.stack.push(lhs + rhs)
+  ? k.cpt.stack.lsCheck(2)
+  let
+    lhs = k.cpt.stack.lsPeekInt(^1)
+    rhs = k.cpt.stack.lsPeekInt(^2)
+  k.cpt.stack.lsShrink(1)
+  k.cpt.stack.lsTop(lhs + rhs)
```
Comment (arnetheduck) [inline nimbus/evm/interpreter/op_handlers/oph_arithmetic.nim:47]: a safer option here would perhaps be to retain the checks, but do something like:
```
template withTop(stack, sym, body) = 
  if stack.len > 0:
    template sym = stack[^1]
    body

withStack(top):
  top = top + rhs
```

Comment (arnetheduck) [reply nimbus/evm/interpreter/op_handlers/oph_arithmetic.nim:47]: then the whole operation becomes:
```
let a = ?stack.popInt(0)
? stack.withTop(top):
  top = top + a
```

Comment (jangko) [reply nimbus/evm/interpreter/op_handlers/oph_arithmetic.nim:47]: Thanks for  the `template sym` trick.

**Comment by arnetheduck** [comment]
see also https://github.com/status-im/nimbus-eth1/issues/2452 - the improvement here will be very contract-dependent - later contracts have a lot more arithmetic in them as they start doing actual cryptography in the contract and these kinds of optimizations become a lot more valuable.

File: nimbus/evm/interpreter/op_handlers/oph_arithmetic.nim (hunk 1 above)
Comment (jangko) [reply nimbus/evm/interpreter/op_handlers/oph_arithmetic.nim]: Both unary and binary ops are optimized with specialized templates.
https://github.com/status-im/nimbus-eth1/pull/2502/commits/a26a16533b2f0202aa5016601a721fb375310c74

**Pull request merged**